workspaces/ui-v2/src/lib/shape-trail-parser.ts

```diff
@@ -73,10 +66,9 @@ export async function shapeTrailParserLastId(
       return {
         lastObject: lastObjectOption
           ? lastObjectOption[0]
-          : field.parentObjectId,
+          : fieldTrail.parentObjectShapeId,
```

Comment:
Could this just always use the trail now?

Reply:
this is needed for new / nested fields.